docs/serving/samples/cloudevents/cloudevents-rust/README.md

```diff
@@ -0,0 +1,111 @@
+   send to a Service or Broker when not initiated by a Broker invocation (e.g.
+   implementing an event source).
+   The input event is modified assigning a new source and type attribute.
+
+The application will use `$K_SINK`-mode whenever the environment variable is
+specified.
+
+Follow the steps below to create the sample code and then deploy the app to your
+cluster. You can also download a working copy of the sample, by running the
+following commands:
+
+```shell
+git clone -b "{{< branch >}}" https://github.com/knative/docs knative-docs
+cd knative-docs/docs/serving/samples/cloudevents/cloudevents-rust
+```
+
+## Before you begin
+
+- A Kubernetes cluster with Knative installed and DNS configured. Follow the
+  [installation instructions](../../../../install/README.md) if you need to
+  create one.
+- [Docker](https://www.docker.com) installed and running on your local machine,
+  and a Docker Hub account configured (we'll use it for a container registry).
+- This guide uses Rust musl toolchain to build the image in order to create
+  really small docker images. To install the Rust toolchain: [`rustup`](https://rustup.rs/).
+  To install musl support: [MUSL support for fully static binaries](https://doc.rust-lang.org/edition-guide/rust-2018/platform-and-target-support/musl-support-for-fully-static-binaries.html).
+
+## Build and deploy the sample
+
+To build the binary, run:
+
+```shell
+cargo build --target x86_64-unknown-linux-musl --release
+```
+
+This will build a statically linked binary, in order to create an image from scratch. Now build the docker image:
+
+```shell
+docker build -t <image> .
+```
+
+To deploy the Knative Service, look in the `service.yaml` and replace `<image>` with the deployed image name. Then run:
+
+```shell
+kubectl apply -f service.yaml
+```
+
+## Testing the sample
+
+Get the URL for your Service with:
+
+```shell
+$ kubectl get ksvc
+NAME                URL                                            LATESTCREATED             LATESTREADY               READY   REASON
+cloudevents-rust    http://cloudevents-rust.xip.io                 cloudevents-rust-vl8fq    cloudevents-rust-vl8fq    True
+```
+
+Then send a CloudEvent to it with:
+
+```shell
+$ curl \
+    -X POST \
+    -H "content-type: application/json"  \
+    -H "ce-specversion: 1.0"  \
+    -H "ce-source: http://curl-command"  \
+    -H "ce-type: curl.demo"  \
+    -H "ce-id: 123-abc"  \
+    -d '{"name":"Dave"}' \
+    http://cloudevents-rust.xip.io
+```
+
+You can also send CloudEvents spawning a temporary curl pod in your cluster with:
+
+```shell
+$ kubectl run curl \
+    --image=curlimages/curl --rm=true --restart=Never -ti -- \
+    -X POST \
+    -H "content-type: application/json"  \
+    -H "ce-specversion: 1.0"  \
+    -H "ce-source: http://curl-command"  \
+    -H "ce-type: curl.demo"  \
+    -H "ce-id: 123-abc"  \
+    -d '{"name":"Dave"}' \
+    http://cloudevents-rust.default.svc
+```
+
+You'll get as result:
```

Comment:
Is this a log message or an HTTP response?

Reply:
This is a log message, let me change it to the curl output

Reply:
I updated it to show the curl verbose output